tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -29,15 +45,327 @@ policy_continuous_aggregate_get_mat_hypertable_id(const Jsonb *config)
 	return mat_hypertable_id;
 }
 
+static int64
+cagg_get_from_config(const Dimension *dim, const Jsonb *config, const char *json_label,
+					 bool *isnull)
+{
+	Oid partitioning_type = ts_dimension_get_partition_type(dim);
+	*isnull = false;
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		bool found;
+		int64 lag = ts_jsonb_get_int64_field(config, json_label, &found);
+		if (!found)
+		{
+			*isnull = true;
+			return (Datum) 0;
+		}
+		Oid now_func = ts_get_integer_now_func(dim);
+
+		Assert(now_func);
+
+		return subtract_integer_from_now_internal(lag, partitioning_type, now_func);
+	}
+	else
+	{
+		Datum res = TimestampTzGetDatum(GetCurrentTimestamp());
+		Interval *lag = ts_jsonb_get_interval_field(config, json_label);
+		if (!lag)
+		{
+			*isnull = true;
+			return 0;
+		}
+		switch (partitioning_type)
+		{
+			case TIMESTAMPOID:
+				res = DirectFunctionCall1(timestamptz_timestamp, res);
+				res = DirectFunctionCall2(timestamp_mi_interval, res, IntervalPGetDatum(lag));
+				break;
+			case TIMESTAMPTZOID:
+				res = DirectFunctionCall2(timestamptz_mi_interval, res, IntervalPGetDatum(lag));
+				break;
+
+			case DATEOID:
+				res = DirectFunctionCall1(timestamptz_timestamp, res);
+				res = DirectFunctionCall2(timestamp_mi_interval, res, IntervalPGetDatum(lag));
+				res = DirectFunctionCall1(timestamp_date, res);
+				break;
```

Comment:
Would be good to have a test case for this.

Reply:
added.